libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -325,32 +325,45 @@ void AP_Mount_Siyi::process_packet()
 #endif
             break;
         }
-        _got_firmware_version = true;
 
         // consume and display camera firmware version
-        _cam_firmware_version = {
-            _msg_buff[_msg_buff_data_start+2],      // firmware major version
-            _msg_buff[_msg_buff_data_start+1],      // firmware minor version
-            _msg_buff[_msg_buff_data_start+0]       // firmware revision (aka patch)
+        _fw_version = {
+            .camera = {
+                .major = _msg_buff[_msg_buff_data_start+2],  // firmware major version
+                .minor = _msg_buff[_msg_buff_data_start+1],  // firmware minor version
+                .patch = _msg_buff[_msg_buff_data_start+0],  // firmware revision (aka patch)
+            },
+            .gimbal = {
+                .major = _msg_buff[_msg_buff_data_start+6],  // firmware major version
+                .minor = _msg_buff[_msg_buff_data_start+5],  // firmware minor version
+                .patch = _msg_buff[_msg_buff_data_start+4],  // firmware revision (aka patch)
+            },
+            .received = true,
         };
 
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Mount: SiyiCam fw:%u.%u.%u",
-                (unsigned)_cam_firmware_version.major,          // firmware major version
-                (unsigned)_cam_firmware_version.minor,          // firmware minor version
-                (unsigned)_cam_firmware_version.patch);         // firmware revision
+        _check_firmware_version();
```

Comment:
Looks like the underscore prefix needs updating here.

By the way, any reason to make the method return a bool if it is never used?  I think the warning is good but I suspect we will never block the camera from being used even if it's old.

Reply:
maybe we should run the check_firmware_version() after we've output the current versions to the user... e.g. move the call down 20 lines or so..

Reply:
> Looks like the underscore prefix needs updating here.

Fixed.

> By the way, any reason to make the check_firmware_version() method return a `bool` if it is never used?

Habit. Made a `void`.

> maybe we should run the check_firmware_version() after we've output the current versions to the user

Updated.